app/controllers/api/agencies_controller.rb

```diff
@@ -31,8 +32,17 @@ def set_agencies
         end
     end
 
+    # get centroid coordinates of submitted zip code
+    def user_zip_coordinates(zip)
+      @loc_lat = Geo.zip_lat(zip)
+      @loc_long = Geo.zip_long(zip)
+    end
+
     def serialized_agencies
-      ActiveModelSerializers::SerializableResource.new(@agencies).as_json
+      ActiveModelSerializers::SerializableResource.new(@agencies,
+                                                       loc_lat: @loc_lat,
+                                                       loc_long: @loc_long)
```

Comment:
See previous comment.  As far as the distance computation is concerned it doesn't matter where the incoming user oriented locations come from (zip code or lat/long params) and it is more more straightforward to deal with them as such.  Perhaps user_ is a better prefix than loc_.  I think it will make more sense when you see the lat/long using the same instance variables.

Reply:
addressed in latest commit.  However it has changed to user_location.  User location is
 any object representing a location known by the user, e.g.,zip code, lat/long derived from address, etc.  Todo: Create an object that represents lat/long search params (next PR)